test/gtest/uct/ib/test_dc.cc

```diff
@@ -404,6 +404,37 @@ UCS_TEST_P(test_dc, dcs_ep_purge_pending) {
     EXPECT_EQ(0, iface->tx.stack_top);
 }
 
+UCS_TEST_P(test_dc, rand_dci) {
+    uct_dc_ep_t *ep;
+
+    if (UCS_OK != uct_config_modify(m_iface_config, "DC_TX_POLICY", "rand")) {
+        UCS_TEST_ABORT("Error: cannot enable random DCI policy");
+    }
+    entity *rand_e = uct_test::create_entity(0);
+    m_entities.push_back(rand_e);
+    uct_iface_set_am_handler(rand_e->iface(), 0, am_dummy_handler, NULL, 0);
```

Comment:
I think this test should at least count that all messages are received...
Maybe also add a parameters with different policies to some sublet of tests to increase the coverage?